Tools/autotest/common.py

```diff
@@ -2269,8 +2307,81 @@ def test_arm_feature(self):
                             self.set_rc(interlock_channel, 1000)
                             raise NotAchievedException("Motor interlock was changed while disarmed")
                 self.set_rc(interlock_channel, 1000)
+
+        self.start_subtest("Test all mode arming")
+        self.load_arming_test_mission()
+        try:
+            for mode in self.mav.mode_mapping():
+                self.start_subtest("Mode : %s" % mode)
+                if mode == "FOLLOW":
+                    self.set_parameter("FOLL_ENABLE", 1)
+                if mode in self.NORMAL_ARMABLE_MODES_LIST:
+                    self.progress("Armable mode : %s" % mode)
+                    self.do_set_mode_via_command_long(mode)
+                    self.arm_vehicle()
+                    if not self.disarm_vehicle():
+                        self.progress("Failed to DISARM")
+                        raise NotAchievedException()
```

Comment:
"Failed to DISARM" should go in here, not be a separate progress message.  I guess that dates this PR a bit!

Reply:
hum... that is a letfover from long time ago